website_lazy_load_image/models/ir_ui_view.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2018 Onestein
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+from lxml import etree
+
+
+class IrUiView(models.Model):
+    _inherit = 'ir.ui.view'
+
+    LAZYLOAD_DEFAULT_SRC = 'data:image/gif;base64,R0lGODlhAQABAIAAAP///' \
+                           'wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw=='
+
+    @api.model
+    def render_template(self, template, values=None, engine='ir.qweb'):
+        """Replaces the src attribute with a data-src attribute
+        for all img elements without the 'lazyload-disable' css class.
+        We use LAZYLOAD_DEFAULT_SRC to prevent showing a broken image
+        icon when the JS is not loading yet.
+        """
+        res = super(IrUiView, self).render_template(template, values, engine)
+        website_id = self.env.context.get('website_id')
+        if website_id and not \
+                self.env['website'].browse(website_id).is_publisher():
+            html = etree.HTML(res)
+            imgs = html.xpath(
+                '//main//img[@src][not(hasclass("lazyload-disable"))]'
```

Comment:
@tarteo have you tested this? :)
Which reminds me: would be better to have a unit test. (Expected to fail :stuck_out_tongue_winking_eye:)

Reply:
Yes, it works :stuck_out_tongue_closed_eyes: I'm currently working on the tests.

Reply:
The tests passed :)